### PR TITLE
Update base_claim_validator to reduce duplication

### PR DESCRIPTION
**Pull request by colinbruce**
As part of a presentation on Code Climate, I need to get some screenshots of making things worse, then better...

As a side effect, the base_claim_validator is now a tiny bit better! :)

### CODE REVIEW

File: app/validators/claim/base_claim_validator.rb [hunk 1]
```diff
@@ -103,12 +103,11 @@ def validate_retrial_actual_length
   # must be one of the list of values
   # must be final third if case type is cracked before retrial (cannot be first or second third)
   def validate_trial_cracked_at_third
-    if cracked_case?
-      validate_presence(:trial_cracked_at_third, 'blank')
-      validate_inclusion(:trial_cracked_at_third, Settings.trial_cracked_at_third, 'invalid')
-      if @record&.case_type&.name == 'Cracked before retrial'
-        validate_pattern(:trial_cracked_at_third, /^final_third$/, 'invalid_case_type_third_combination')
-      end
+    return unless cracked_case?
```
Comment (jsugarman) [inline app/validators/claim/base_claim_validator.rb:106]: 👍 clearer

File: app/validators/claim/base_claim_validator.rb [hunk 2]
```diff
@@ -260,28 +258,26 @@ def validate_retrial_length(field)
...
-      @record.first_day_of_trial.present? &&
-      @record.trial_concluded_at.present?
+  def validate_retrial_actual_length_consistency
+    return unless actual_length_consistency_for(requires_retrial_dates?, @record.retrial_actual_length, @record.retrial_started_at, @record.retrial_concluded_at)
+    add_error(:retrial_actual_length, 'too_long')
   end
 
-  def validate_retrial_actual_length_consistency
-    return unless actual_length_consistency_for_retrial
+  def actual_length_consistency_for(requires_dates, actual_length, start_date, end_date)
```
Comment (jsugarman) [inline app/validators/claim/base_claim_validator.rb]: naming suggestion: `actual_length_consistent?` | `actual_length_valid?`

which matches the other method `trial_length_valid?` lower down called by this method

File: app/validators/claim/base_claim_validator.rb [hunk 3]
```diff
@@ -177,13 +174,11 @@ def validate_trial_fixed_notice_at
...
   # cannot be before trial_fixed_notice_at
   def validate_trial_fixed_at
-    return if @record.disable_for_state_transition.eql?(:only_amount_assessed)
-    if @record.case_type && @record.requires_cracked_dates?
-      validate_presence(:trial_fixed_at, 'blank')
-      validate_too_far_in_past(:trial_fixed_at)
-      validate_on_or_after(@record.trial_fixed_notice_at, :trial_fixed_at,
-                           'check_not_earlier_than_trial_fixed_notice_at')
-    end
+    return if ignore_validation_for_trial_dates
```
Comment (jsugarman) [inline app/validators/claim/base_claim_validator.rb]: naming suggestion: would `ignore_validation_for_cracked_trial_dates` be more applicable and descriptive -

**Review by jsugarman — APPROVED** [review]
love the `return if/unless`. just a couple of naming suggestions

File: app/validators/claim/base_claim_validator.rb (hunk 2 above)
Comment (colinbruce) [reply app/validators/claim/base_claim_validator.rb]: Good call on the renamings  "naming things is hard"! ;)

**Pull request merged**